MiniSim/Menu.swift

```diff
@@ -121,103 +121,125 @@ class Menu: NSMenu {
+        guard let header = self.items.first(where: { $0.tag == section.rawValue }),
+              let startIndex = self.items.firstIndex(of: header) else {
+            return
+        }
+        
+        var count = 0
+        items.forEach { menuItem in
+            count += 1
+            if let itemIndex = self.items.firstIndex(where: { $0.title == menuItem.title }) {
+                self.replaceMenuItem(at: itemIndex, with: menuItem)
+                return
             }
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(IOSSubMenuClick)
-            subMenu.addItem(menuItem)
+            self.safeInsertItem(menuItem, at: startIndex + count)
         }
+    }
+
+    private func createMenuItem(for device: Device) -> NSMenuItem {
+        let menuItem = NSMenuItem(
+            title: device.displayName,
+            action: #selector(deviceItemClick),
+            keyEquivalent: "",
+            type: device.platform == .ios ? .launchIOS : .launchAndroid
+        )
         
-        for item in DeviceService.getCustomCommands(platform: .ios) {
-            let menuItem = IOSSubMenuItem.customCommand.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(IOSSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
+        menuItem.target = self
+        menuItem.keyEquivalentModifierMask = [.command]
+        menuItem.submenu = buildSubMenu(for: device)
+        menuItem.state = device.booted ? .on : .off
+        return menuItem
+    }
+    
+    private func replaceMenuItem(at index: Int, with newItem: NSMenuItem) {
+        self.removeItem(at: index)
+        self.insertItem(newItem, at: index)
+    }
+    
+    func buildSubMenu(for device: Device) -> NSMenu {
+        let subMenu = NSMenu()
+        let platform = device.platform
+        let callback = platform == .android ? #selector(androidSubMenuClick) : #selector(IOSSubMenuClick)
+        let actionsSubMenu = createActionsSubMenu(
+            for: platform.subMenuItems,
+            isDeviceBooted: device.booted,
+            callback: callback)
+        let customCommandSubMenu = createCustomCommandsMenu(
+            for: platform,
+            isDeviceBooted: device.booted,
+            callback: callback)
+        (actionsSubMenu + customCommandSubMenu).forEach { subMenu.addItem($0) }
+        return subMenu
+    }
+
+    func createActionsSubMenu(
+        for subMenuItems: [SubMenuItem],
+        isDeviceBooted: Bool,
+        callback: Selector
+    ) -> [NSMenuItem] {
+        subMenuItems.compactMap { item in
+            if item is SubMenuItems.Separator {
+                return NSMenuItem.separator()
             }
-            if item.bootsDevice ?? false && booted {
-                continue
+            else if let item = item as? SubMenuActionItem {
```

Comment:
You can just use if let without else and also `if let item as? SubMenuActionItem`

Reply:
Removed `else`, but it still requires assignment.